### PR TITLE
Doceree AdManager Bid Adapter: initial release 

### PR DESCRIPTION
**Pull request by Doceree-techStack**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [x] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes

- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

### CODE REVIEW

**Comment by Doceree-techStack** [comment]
Please update the status of (https://github.com/prebid/Prebid.js/pull/10684) pull request

**Comment by Doceree-techStack** [comment]
Please update the status of (https://github.com/prebid/Prebid.js/pull/10684) pull request

**Comment by Doceree-techStack** [comment]
added the link to docs PR

**Comment by Doceree-techStack** [comment]
@osazos please review to move the pull request for merge as we have urgent integration blocked

**Review by osazos — COMMENTED** [review]
When testing your endpoint:

Your  `serverResponse.body` response is an array but your code is written to handle an object. Btw I didn't receive a "valid" response so perhaps it is an exepected format.

Also, it seems there is a mismatch regarding the `requestId` in the request and the response:

```
// request
{
  "userid": "7d26d8ca-233a-46c2-9d36-7c5d261e151d",
  ... 
  "requestId": "296180d3d4c11b",
}

// response
{
  "response": [
    {
      "status": -1,
      "adUnit": "DOC_7jm9j5eqkl0xvc5w",
      "position": 0,
      "adFormat": 0,
      "userId": "7d26d8ca-233a-46c2-9d36-7c5d261e151d",
      "bidRequestId": "0bdf4e05-38cf-43c1-8dca-8f1ed0a68e01-7sjmhc"
    }
  ]
}
```

Which diplay the warning: 
```
 WARNING: Bidder docereeadmanager made bid for unknown request ID: undefined. Ignoring.
```

**Comment by Doceree-techStack** [comment]
@osazos the changes you mentioned have been fixed. Kindly have a look on the updated code and make sure to use an updated adUnit code for making a server call.

Please let us know if we need to change anything else.

File: modules/docereeAdManager.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+  isGdprConsentPresent: (bid) => {
+    const { gdpr, gdprconsent } = bid.params;
+    if (gdpr == '1') {
+      return !!gdprconsent;
+    }
+    return true;
+  },
+  buildRequests: (validBidRequests) => {
+    const serverRequests = [];
+    const { data } = config.getConfig('docereeadmanager.user');
```
[the remark this answers is not in the document]
Comment (osazos) [reply modules/docereeAdManager.js]: In fact this is useless, my bad. The issue is you don't check in `getPayload()` if the `userData` param exists.

File: modules/docereeAdManager.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+          contentType: 'application/json',
+          withCredentials: true,
+        },
+      });
+    });
+
+    return serverRequests;
+  },
+  interpretResponse: (serverResponse) => {
+    const responseJson = serverResponse ? serverResponse.body : {};
```
[the remark this answers is not in the document]
Comment (osazos) [reply modules/docereeAdManager.js]: I get a "valid" response but you still don't handle the case your response is not an object. I got this case when I use a bad `placementId`:

```
placementId: "DOC-500-3", //required
```

File: modules/docereeAdManager.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (Doceree-techStack) [reply modules/docereeAdManager.js]: @osazos Issue resolved please check

[the remark this answers is not in the document]
Comment (osazos) [reply modules/docereeAdManager.js]: Good to me for this one.
Plz check about the last issue : https://github.com/prebid/Prebid.js/pull/10684#discussion_r1390764623

File: modules/docereeAdManager.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (osazos) [reply modules/docereeAdManager.js]: Thx

**Review by osazos — APPROVED** [review]
LGTM

**Pull request merged**